### PR TITLE
Upgraded to Play 2.4.0-RC1

### PR DESCRIPTION
**Pull request by jroper**
Also:

* Removed MailerPlugin classes
* Moved MailerJavaClient to play.libs.mailer.MailerClient
* Provided bindings for play.libs.mailer.MailerClient
* Documented how to inject a MailerClient
* Added tests for the mailer module

### CODE REVIEW

**Comment by Mogztter** [comment]
Build failed because of an unresolved dependency:
```
unresolved dependency: org.scalaz.stream#scalaz-stream_2.11;0.7a
```

**Comment by Mogztter** [comment]
> Removed MailerPlugin classes

@vmouta will be sad, see #31

**Comment by jroper** [comment]
Oh.  I'll reinstate.

**Comment by duketon** [comment]
@jroper are there any plans to go ahead with this PR? I'd like to base off this and contribute some integrations with SendGrid and other providers.

**Comment by Mogztter** [comment]
For the unresolved dependency someone suggested to add the bintray repository: https://github.com/etorreborre/specs2/issues/347#issuecomment-90262534

Apart from that this is looking good to me.

@duketon cool ! I've done some work with Mandrill if you are interested.

**Comment by duketon** [comment]
@Mogztter Sure, sounds good! I'm currently using the SendGrid Java SDK in a project, and wrapped all operations in Future blocks. I'm happy with the way it's been working. Although in this plugin it might be more sensible to use provider's Web APIs instead of adding a dependency/ies for each possible client. Let me know what you think :-)

**Comment by jroper** [comment]
Sorry forgot about this. I've updated the PR, reinstated PlayMailer, upgraded to 2.4.0-RC3, added the resolver.

**Comment by mkurz** [comment]
@jroper Build fails:
> [error] /home/travis/build/playframework/play-mailer/src/main/scala/play/api/libs/mailer/MailerPlugin.scala:25: @ deprecated now takes two arguments; see the scaladoc.

File: src/main/scala/play/api/libs/mailer/MailerPlugin.scala [hunk 1]
```diff
@@ -7,19 +7,29 @@ import javax.mail.internet.InternetAddress
...
+    bind[JMailerClient].to(bind[MailerClient]),
+    bind[MailerClient].qualifiedWith("mock").to[MockMailer],
+    bind[JMailerClient].qualifiedWith("mock").to[MockMailer]
   )
 }
 
 // API
 
-trait MailerClient extends MailerJavaClient {
+@deprecated("Use injected MailerClient instead")
```
Comment (Mogztter) [inline src/main/scala/play/api/libs/mailer/MailerPlugin.scala]: Since 2.4.0 ?

**Comment by Mogztter** [comment]
@duketon yes we don't want to add dependencies for each providers. You can use the Play WS API to make HTTP requests to provider's web API.

**Comment by jroper** [comment]
Build is green.

**Comment by Mogztter** [comment]
:+1:

**Comment by mkurz** [comment]
@Mogztter Some changes (like #25 and #28) are only in the `master` branch but not in the `play24` (yet). Is this intentional? Will the branches get merged or the changes be cherry-picked at some point?

**Comment by Mogztter** [comment]
@Mkurz no this is not intentonial, we must cherry-pick the changes. Maybe we should track these tasks as new issues to remember what need to be done ?

**Comment by mkurz** [comment]
We can do that (Maybe also introduce milestones? But not sure about that.).
I also propose that (at the very latest when Play 2.4 is released) the `play24` becomes the `master` branch and the current `master` branch is renamed to `2.3.x` (or similiar). (People will mostly PR against the `master'`)

**Comment by mkurz** [comment]
I added a new issue collecting the changes that need to be done: #38

**Comment by duketon** [comment]
:+1:

**Comment by jroper** [comment]
I agree, it's time the play24 branch became the master branch, and master 2.3.x.

**Comment by Mogztter** [comment]
> I agree, it's time the play24 branch became the master branch, and master 2.3.x.

Ok I will make that change

**Pull request merged**